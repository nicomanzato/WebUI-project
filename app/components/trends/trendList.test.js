import React from 'react';
import {TouchableHighlight} from 'react-native'
import TrendList from './trendList'
import TrendMock from './../../store/trend/mock/trendMock'
import { shallow, mount, ReactWrapper } from 'enzyme'

describe('trend list', () => {

  it('should handle dispatching its actions', () => {
    const onItemPress = jest.fn().mockName('mockedFunction');

    const wrapper = mount(
      <TrendList
        trends={[TrendMock, TrendMock, TrendMock]}
        onItemPress={onItemPress}
      />
    );

    const trend = wrapper.find('Trend').first().find('TouchableHighlight').first();
    trend.props().onPress();

    expect(onItemPress).toHaveBeenCalled();
  });

  it('should render correctly', () => {
    const wrapper = shallow(
      <TrendList
        trends={[TrendMock, TrendMock, TrendMock]}
        onItemPress={() => {}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

});
