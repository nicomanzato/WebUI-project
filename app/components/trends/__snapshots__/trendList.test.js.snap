// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`trend list should render correctly 1`] = `
<ScrollView>
  <Trend
    index={0}
    key="0"
    onPress={[Function]}
    trend={
      Object {
        "name": "Sofía Herrera",
        "promoted_content": null,
        "query": "%22Sof%C3%ADa+Herrera%22",
        "tweet_volume": 11111,
        "url": "http://twitter.com/search?q=%22Sof%C3%ADa+Herrera%22",
      }
    }
  />
  <Trend
    index={1}
    key="1"
    onPress={[Function]}
    trend={
      Object {
        "name": "Sofía Herrera",
        "promoted_content": null,
        "query": "%22Sof%C3%ADa+Herrera%22",
        "tweet_volume": 11111,
        "url": "http://twitter.com/search?q=%22Sof%C3%ADa+Herrera%22",
      }
    }
  />
  <Trend
    index={2}
    key="2"
    onPress={[Function]}
    trend={
      Object {
        "name": "Sofía Herrera",
        "promoted_content": null,
        "query": "%22Sof%C3%ADa+Herrera%22",
        "tweet_volume": 11111,
        "url": "http://twitter.com/search?q=%22Sof%C3%ADa+Herrera%22",
      }
    }
  />
</ScrollView>
`;
